packages/cli/src/imagerenderer.ts

```diff
@@ -0,0 +1,149 @@
+import sharp from "sharp";
+
+export type TerminalType = "kitty" | "iterm2" | "none";
+
+const KITTY_IDENTIFIERS: string[] = [
+  "kitty",
+  "wezterm",
+  "konsole",
+  "warp",
+  "wayst",
+  "st",
+];
+
+type KittyCommand = Record<string, string | number>;
+
+export function detectTerminalCapabilities(): TerminalType {
+  const termProgram = (Deno.env.get("TERM_PROGRAM") || "").toLowerCase();
+
+  for (const id of KITTY_IDENTIFIERS) {
+    if (termProgram === id) {
+      return "kitty";
+    }
+  }
+
+  if (termProgram === "iterm.app") {
+    return "iterm2";
+  }
+
+  return "none";
+}
+
+function serializeGrCommand(
+  cmd: KittyCommand,
+  payload?: string,
+): Uint8Array {
+  const cmdString = Object.entries(cmd)
+    .map(([k, v]) => `${k}=${v}`)
+    .join(",");
+
+  const encoder = new TextEncoder();
+  const parts: Uint8Array[] = [];
+
+  parts.push(encoder.encode("\x1b_G"));
+  parts.push(encoder.encode(cmdString));
+
+  if (payload) {
+    parts.push(encoder.encode(";"));
+    parts.push(encoder.encode(payload));
+  }
+
+  parts.push(encoder.encode("\x1b\\"));
+
+  const totalLength = parts.reduce((sum, part) => sum + part.length, 0);
+  const result = new Uint8Array(totalLength);
+  let offset = 0;
+
+  for (const part of parts) {
+    result.set(part, offset);
+    offset += part.length;
+  }
+
+  return result;
+}
+
+export async function renderImageKitty(
+  imagePath: string,
+  cmd: KittyCommand,
+): Promise<void> {
+  const data = await Deno.readFile(imagePath);
+  const base64Data = btoa(String.fromCharCode(...data));
```

Comment:
The `btoa()` function is deprecated.  Use the [`encodeBase64()`](https://jsr.io/@std/encoding/doc/~/encodeBase64) function from the [@std/encoding](https://jsr.io/@std/encoding) package.  You can install it using `deno add` command:

```sh
deno add jsr:@std/encoding
```